src/en/azuki/src/eu/kanade/tachiyomi/extension/en/azuki/Azuki.kt

```diff
@@ -0,0 +1,228 @@
+            val link = chapterRow.selectFirst("a.a-card-link") ?: return@mapNotNull null
+            val href = link.attr("href")
+            var chapterId: String? = null
+
+            if (href.startsWith("/checkout/")) {
+                chapterId = href.substringAfter("chapter_uuids%5B%5D=").substringBefore("&")
+            } else if (href.contains("/read/")) {
+                chapterId = href.substringAfter("/read/")
+            }
+
+            if (chapterId.isNullOrEmpty()) return@mapNotNull null
+
+            SChapter.create().apply {
+                url = "/series/$seriesSlug/read/$chapterId"
+                name = link.selectFirst(".m-chapter-row__title-cluster span")?.text() ?: link.text()
+                date_upload = chapterRow.selectFirst(".m-chapter-row__date time")?.attr("datetime")
+                    ?.let { dateFormat.tryParse(it) } ?: 0L
+
+                val isPremium = chapterRow.selectFirst(".m-chapter-row__premium-badge") != null ||
+                    chapterRow.parent()?.hasClass("m-chapter-card--secondary") == true
+
+                if (isPremium && chapterId !in unlockedChapterIds) {
+                    name = "🔒 $name"
+                }
+            }
+        }.reversed()
+    }
+
+    // Pages
+    override fun pageListRequest(chapter: SChapter): Request {
+        val chapterId = chapter.url.substringAfter("/read/")
+        val apiUrl = "$apiUrl/chapters/$chapterId/pages/v1"
+        return GET(apiUrl, apiHeaders())
+    }
+
+    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {
+        return client.newCall(pageListRequest(chapter))
+            .asObservable()
+            .map { response ->
+                if (!response.isSuccessful) {
+                    if (response.code == 401 || response.code == 403) {
+                        throw Exception("This chapter is locked. Log in and unlock the chapter to view.")
+                    }
+                    throw Exception("HTTP error ${response.code}")
+                }
+                pageListParse(response)
+            }
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val result = response.parseAs<PageListDto>()
+        return result.data!!.pages.mapIndexed { i, page ->
+            val imageList = page.image!!.webp ?: page.image.jpg ?: throw Exception("No images found for page ${i + 1}")
+
+            val bestAvailableUrl = imageList.maxByOrNull { it.width }?.url
+                ?: throw Exception("No image URL found for page ${i + 1}")
+
+            val resolutionRegex = Regex("""/(\d+)\.(webp|jpg)$""")
+            val match = resolutionRegex.find(bestAvailableUrl)
+
+            val imageUrl: String
+            if (match != null && match.groupValues[1].toInt() < 1600) {
+                val highResUrl = resolutionRegex.replace(bestAvailableUrl, "/1600.${match.groupValues[2]}")
+                imageUrl = "$highResUrl?drm=1, $bestAvailableUrl?drm=1"
```

Comment:
do you handle these multiple image urls anywhere? otherwise having two urls comma separated won't work